onnxruntime/test/shared_lib/test_inference.cc

```diff
@@ -2178,6 +2251,101 @@ TEST(CApiTest, io_binding_cuda) {
 }
 #endif
 
+#if defined(USE_QNN)
+
+TEST(CApiTest, io_binding_qnn_htp_shared) {
+  Ort::Session session{nullptr};
+  if (!CreateSessionWithQnnEpAndQnnHtpSharedMemoryAllocator(MODEL_URI, session)) {
+    GTEST_SKIP() << "HTP shared memory allocator is unavailable.";
+  }
+
+  Ort::MemoryInfo info_qnn_htp_shared("QnnHtpShared", OrtAllocatorType::OrtDeviceAllocator, 0, OrtMemTypeDefault);
+
+  Ort::Allocator qnn_htp_shared_allocator(session, info_qnn_htp_shared);
+  auto allocator_info = qnn_htp_shared_allocator.GetInfo();
+  ASSERT_EQ(info_qnn_htp_shared, allocator_info);
+
+  const std::array<int64_t, 2> x_shape = {3, 2};
+  std::array<float, 3 * 2> x_values = {1.0f, 2.0f, 3.0f, 4.0f, 5.0f, 6.0f};
+  auto input_data = qnn_htp_shared_allocator.GetAllocation(x_values.size() * sizeof(float));
```

Comment:
>input_data [](http://example.com/codeflow?start=7&length=10)

Is this Freed() some place?
Perhaps, as an example, we could wrap it into some sort of smart_ptr?

Reply:
yes, the returned `MemoryAllocation` object should take care of freeing it.
https://github.com/microsoft/onnxruntime/blob/80d8931f1d835fe29d89cced25f05be6690f81b6/include/onnxruntime/core/session/onnxruntime_cxx_inline.h#L183-L189